src/tvheadend/AutoRecordings.cpp

```diff
@@ -174,8 +175,15 @@ PVR_ERROR AutoRecordings::SendAutorecAddOrUpdate(const kodi::addon::PVRTimer& ti
 
   htsmsg_add_str(m, "name", timer.GetTitle().c_str());
 
-  /* epg search data match string (regexp) */
-  htsmsg_add_str(m, "title", timer.GetEPGSearchString().c_str());
+  /* epg search data match string */
+  std::string searchString = timer.GetEPGSearchString();
+  if (!Settings::GetInstance().GetAutorecUseRegEx())
+  {
+    // escape regex special chars
+    static const std::regex specialChars(R"([-[\]{}()*+?.,\^$|#])");
```

Comment:
Have you validated that this regular expression does what you want it to do?

Reply:
Yes. Do you have concerns?

Reply:
I found it on stackoverflow.

Reply:
I had some, but when I looked closer I think they're irrelevant. Mainly about escaping `-`, but I think it's okay in its current form.

Reply:
You may be right. Shall I remove - from the list?

Reply:
And what about the `,`?

Reply:
Reading a bit further, I think I will let it like it is now.